synapse/storage/databases/main/metrics.py

```diff
@@ -74,9 +87,26 @@ def __init__(
     ):
         super().__init__(database, db_conn, hs)
 
-        # Read the extrems every 60 minutes
         if hs.config.worker.run_background_tasks:
-            self._clock.looping_call(self._read_forward_extremities, 60 * 60 * 1000)
+            # Read the extrems every 60 minutes
+            self._clock.looping_call(self._read_forward_extremities, HOUR_IN_MS)
+            # update room metrics every 15 minutes
+            self._clock.looping_call(self.set_known_rooms_gauge, QUARTER_HOUR_IN_MS)
+            self._clock.looping_call(
+                self.set_locally_joined_rooms_gauge, QUARTER_HOUR_IN_MS
+            )
```

Comment:
Not sure if it matters, but maybe you could put both into the same call, i.e.:

```suggestion
            def update_room_metrics():
                self.set_known_rooms_gauge()
                self.set_locally_joined_rooms_gauge()
                
            self._clock.looping_call(update_room_metrics, QUARTER_HOUR_IN_MS)
```

That would then also possibly simplify the setup thing, not sure if we need to run that as a background process there :)